hystrix-contrib/hystrix-javanica/src/main/java/com/netflix/hystrix/contrib/javanica/aop/aspectj/HystrixCommandAspect.java

```diff
@@ -168,33 +167,38 @@ public MetaHolder create(Object proxy, Method collapserMethod, Object obj, Objec
             }
 
             Method batchCommandMethod = getDeclaredMethod(obj.getClass(), hystrixCollapser.batchMethod(), List.class);
-            if (batchCommandMethod == null || !batchCommandMethod.getReturnType().equals(List.class)) {
+
+            if (batchCommandMethod == null)
+                throw new IllegalStateException("batch method is absent: " + hystrixCollapser.batchMethod());
+
+            Class<?> batchReturnType = batchCommandMethod.getReturnType();
+            Class<?> collapserReturnType = collapserMethod.getReturnType();
+            boolean observable = collapserReturnType.equals(Observable.class);
+
+            if (!batchReturnType.equals(List.class))
```

Comment:
I guess this check is redundant because
 ```getDeclaredMethod(obj.getClass(), hystrixCollapser.batchMethod(), List.class)``` already checks return type, if batch method return type isn't List then ```getDeclaredMethod``` returns null.